reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/InternalSdkCoreContextTest.kt

```diff
@@ -0,0 +1,626 @@
+import fr.xgouchet.elmyr.annotation.StringForgeryType
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import fr.xgouchet.elmyr.jvm.useJvmFactories
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Offset
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * Provides the tests for the InternalSdkCore API not related with the writing operations.
+ */
+@RunWith(AndroidJUnit4::class)
+class InternalSdkCoreContextTest : MockServerTest() {
+
+    @get:Rule
+    var forge = ForgeRule().useJvmFactories().useToolsFactories().withFactory(ConfigurationCoreForgeryFactory())
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeUserId: String
+
+    @StringForgery(regex = "[A-Z][a-z]+ [A-Z]\\. [A-Z][a-z]+")
+    lateinit var fakeUserName: String
+
+    @StringForgery(regex = "[a-z]+\\.[a-z]+@[a-z]+\\.[a-z]{3}")
+    lateinit var fakeUserEmail: String
+    private var fakeUserAdditionalProperties: Map<String, Any?> = emptyMap()
+    private lateinit var stubFeature: StorageBackedFeature
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeFeatureName: String
+
+    private lateinit var fakeTrackingConsent: TrackingConsent
+
+    private var featureSdkCore: FeatureSdkCore? = null
+    private var testedInternalSdkCore: InternalSdkCore? = null
+
+    @Forgery
+    lateinit var fakeConfiguration: Configuration
+    private var currentDeviceTimeInMs: Long = 0L
+
+    @Before
+    fun setUp() {
+        currentDeviceTimeInMs = System.currentTimeMillis()
+        stubFeature = StubStorageBackedFeature(
+            forge,
+            fakeFeatureName,
+            getMockServerWrapper().getServerUrl()
+        )
+        fakeTrackingConsent = forge.aValueFrom(TrackingConsent::class.java)
+        fakeUserAdditionalProperties = forge.exhaustiveAttributes(excludedKeys = setOf("id", "name", "email"))
+        testedInternalSdkCore = Datadog.initialize(
+            ApplicationProvider.getApplicationContext(),
+            fakeConfiguration,
+            fakeTrackingConsent
+        ) as? InternalSdkCore
+        Datadog.setUserInfo(fakeUserId, fakeUserName, fakeUserEmail, fakeUserAdditionalProperties)
+        featureSdkCore = testedInternalSdkCore
+        featureSdkCore?.registerFeature(stubFeature)
+    }
+
+    @After
+    fun tearDown() {
+        Datadog.stopInstance()
+    }
+
+    // region get Datadog Context
+
+    @Test
+    fun must_returnCorrectDatadogContext_when_getDatadogContext() {
+        // Given
+        val expectedServiceName = fakeConfiguration.service() ?: PACKAGE_NAME
+
+        // When
+        val context = testedInternalSdkCore?.getDatadogContext()
+
+        // Then
+        assertThat(context?.sdkVersion).isEqualTo(BuildConfig.SDK_VERSION_NAME)
```

Comment:
having `checkNotNull(context)` will also unlock smart cast for the code below, removing the need to add `.?`

Same for other similar tests

```suggestion
        // Then
        checkNotNull(context)
        assertThat(context.sdkVersion).isEqualTo(BuildConfig.SDK_VERSION_NAME)
```